Automattic/woocommerce-payments
# Add dispute steps to resolve to the transaction details page.

Fixes https://github.com/Automattic/woocommerce-payments/issues/6925.

> **Note**
> This PR is behind feature flag `_wcpay_feature_dispute_on_transaction_page`

#### Changes proposed in this Pull Request

Add the 'Steps to resolve' section on the transaction details page for disputes (**not inquiries** – handled in issue #7245).

![steps](https://github.com/Automattic/woocommerce-payments/assets/3147296/52335a9a-b2e9-485e-a0c2-45de19daad74)

**Inquiries** will not see the steps rendered:
![image](https://github.com/Automattic/woocommerce-payments/assets/3147296/1bb193ea-ead7-407e-95c6-21ca6dd8fb60)


Notes:
- The step **Review the claim** is not part of this PR's scope – now handled in issue #7156.
- The text 'guidance on dispute withdrawal' from the third step is supposed to link to a doc ~but it's not yet ready.~ This has been published.
- 

#### Testing instructions

* Enable the feature flag by running `update_option( '_wcpay_feature_dispute_on_transaction_page', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
* Purchase a product with card `4000000000000259` to get disputed right away.
* Open the transaction details page of that new order by going to admin > Payments > Transactions > click on the top most transaction.
* Confirm there is 'Steps to resolve' section.
* Confirm the steps are:
```
1. Email the customer to address their concerns.
2. Provide guidance on dispute withdrawal if the customer agrees.
3. Challenge (?) or accept (?) the dispute by {dispute due by date} ({n} days left to respond)
```
* Click on the 'Email the customer' link. It should open your local email client. Confirm:
  * To address is the customer's email address.
  * Subject is `Problem with your purchase from {store name} on {YYYY-MM-DD of the charge date}?`
  * Body is:
```
Hello {customer name}

We noticed that on {YYYY-MM-DD of the dispute created date}, you disputed a {currency formatted dispute amount, eg: $10.00} from {YYYY-MM-DD of the charge date}. We wanted to contact you to make sure everything was all right with your purchase and see if there's anything else we can do to resolve any problems you might have had.

Alternatively, if the dispute was a mistake, you could easily withdraw it by calling the number on the back of your card. Thank you so much - we appreciate your business and look forward to working with you.
```
* Confirm the 2nd step has 'guidance on dispute withdrawal' link to https://woocommerce.com/document/woopayments/fraud-and-disputes/managing-disputes/#withdrawals.
* Confirm the 3rd step has icons:
  * The first icon when clicked will explain about challenging the dispute:
```
Challenge the dispute if you consider the claim invalid. You'll need to provide evidence to back your claim. Keep in mind that challenging doesn't ensure a resolution in your favor.
```
  * The second icon when clicked will explain about accepting the dispute:
```
Accepting this dispute will automatically close it. Your account will be charged a {currency formatted dispute fee, eg: $15.00} fee, and the disputed amount will be refunded to the cardholder.
```
* Confirm the 3rd step shows the dispute's due by date:
  * You can either edit [this code](https://github.com/Automattic/woocommerce-payments/blob/04ee3f67f48e6bb5dcf3deb34986fc89f879ebf8/client/payment-details/dispute-details/dispute-steps.tsx#L156-L160) or pass `daysRemaining` at your will [here](https://github.com/Automattic/woocommerce-payments/blob/04ee3f67f48e6bb5dcf3deb34986fc89f879ebf8/client/payment-details/dispute-details/index.tsx#L70), or hardcode the dispute's due by date to test these cases:
  * When due date is still 7-day away or more, the text color is black.
		<img width="643" alt="Screenshot 2023-09-14 at 06 59 20" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/e64a833a-dcc1-45ce-8329-c6c5be4c6dd4">
  * When due date is between 2 and 7-day away, the text color is yellow.
		<img width="647" alt="Screenshot 2023-09-14 at 07 02 53" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/766271ac-1d39-4fd0-86c1-0abe0ad7f081">
  * When due date is less than 3-day away, the text color is red.
		<img width="650" alt="Screenshot 2023-09-14 at 07 04 36" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/2af53f9a-a770-41dc-9888-f30c9622017a">
  * When due date is today, the text color is red and it says `Last day today`.
		<img width="598" alt="Screenshot 2023-09-14 at 07 06 45" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/7c06d928-2a25-4a59-a829-12f2400fd743">

**Inquiries**
* Purchase a product with card `4000000000001976` to create an inquiry.
* Open the transaction details page of that new order by going to admin > Payments > Transactions > click on the top most transaction.
* Confirm that the card is rendered correctly with the appropriate styling, but no "Steps to Resolve" section is displayed.


-------------------


**Post merge**
